stm32f756zg

Dear community!

Would it be possible to add support for 1MB flash on [NUCLEO-F756ZG
](https://www.st.com/en/evaluation-tools/nucleo-f756zg.html)? While there is no support, it's not possible to use an update feature via mcuboot.

I tried to do it by myself, but didn't work like I'm expecting.
```
&flash0 {

	partitions {
		compatible = "fixed-partitions";
		#address-cells = <1>;
		#size-cells = <1>;

		/*
		 * The two sectors 0-1 (32+32 kbytes) are reserved for
		 * the bootloader.
		 */
		boot_partition: partition@0 {
			label = "mcuboot";
			reg = <0x0 DT_SIZE_K(64)>;
			read-only;
		};;

		/*
		 * The flash starting at 0x00010000 and ending at
		 * 0x0001ffff (sectors 16-31) is reserved for use
		 * by the application.
		 */
		storage_partition: partition@1e000 {
			label = "storage";
			reg = <0x0001e000 0x00002000>;
		};

		slot0_partition: partition@20000 {
			label = "image-0";
			reg = <0x00020000 0x00060000>;
		};
		slot1_partition: partition@80000 {
			label = "image-1";
			reg = <0x00080000 0x00060000>;
		};
		scratch_partition: partition@e0000 {
			label = "image-scratch";
			reg = <0x000e0000 0x00020000>;
		};
	};
};
```
After that boards booted, but there is a message `Cannot upgrade: slots have non-compatible sectors`... .
```
uart:~$ *** Booting Zephyr OS build v3.4.0-rc1-90-g5c7d2f1a90a6 ***
I: Starting bootloader
W: Cannot upgrade: slots have non-compatible sectors
I: Bootloader chainload address offset: 0x20000
�: Jumping to the first image slot

*** Booting Zephyr OS build v3.4.0-rc1-90-g5c7d2f1a90a6 ***
[00:00:00.011,000] <inf> main: Hawkbit sample app started
[00:00:00.011,000] <inf> main: Image build time: Jun 26 2023 15:19:
```
At the end, I was trying to get the `Hawkbit` client running, but it didn't init the network at all. 
So what I'm doing wrong?

Thanks

I try to build the `hawkbit` with `--sysbuild`, but it failed:
```
west build -b nucleo_f756zg --sysbuild

[4/321] Generating include/generated/version.h
-- Zephyr version: 3.4.0-rc1 (/home/vboxuser/zephyrproject/zephyr), build: v3.4.0-rc1-90-g5c7d2f1a90a6
[311/321] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit/zephyr/zephyr_pre0.map 
: && ccache /opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/newlib/liblib__libc__newlib.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/subsys/random/libsubsys__random.a  zephyr/subsys/dfu/boot/libsubsys__dfu__boot.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ethernet/libsubsys__net__l2__ethernet.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/dns/libsubsys__net__lib__dns.a  zephyr/subsys/net/lib/http/libsubsys__net__lib__http.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/flash/libdrivers__flash.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/hwinfo/libdrivers__hwinfo.a  zephyr/drivers/ethernet/libdrivers__ethernet.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/reset/libdrivers__reset.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a  modules/mbedtls/libmbedTLSBase.a  modules/mbedtls/libmbedTLSCrypto.a  modules/mbedtls/libmbedTLSX509.a  modules/mbedtls/libmodules__mbedtls.a  modules/mcuboot/libmcuboot_util.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/thumb/v7e-m/nofp"  -L/home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  modules/mbedtls/libmbedTLSBase.a  modules/mbedtls/libmbedTLSCrypto.a  modules/mbedtls/libmbedTLSX509.a  -mcpu=cortex-m7  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -lm  -Wl,-lc  -L"/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/arm-zephyr-eabi"/lib/thumb/v7e-m/nofp  -Wl,-lgcc  -lc && cd /home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit/zephyr && /usr/bin/cmake -E echo
/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `main':
/home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/src/main.c:37: undefined reference to `hawkbit_init'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FAILED: hawkbit-prefix/src/hawkbit-stamp/hawkbit-build /home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit-prefix/src/hawkbit-stamp/hawkbit-build 
cd /home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/hawkbit && /usr/bin/cmake --build .
ninja: build stopped: subcommand failed.
```
But if I build it without it, it works fine:
```
west build -b nucleo_f756zg

[1/329] Preparing syscall dependency handling

[2/329] Generating include/generated/version.h
-- Zephyr version: 3.4.0-rc1 (/home/vboxuser/zephyrproject/zephyr), build: v3.4.0-rc1-90-g5c7d2f1a90a6
[319/329] Linking C executable zephyr/zephyr_pre0.elf

[323/329] Linking C executable zephyr/zephyr_pre1.elf

[329/329] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:      181912 B       256 KB     69.39%
             RAM:       67328 B       256 KB     25.68%
            QSPI:          0 GB       256 MB      0.00%
            DTCM:       12544 B        64 KB     19.14%
        IDT_LIST:          0 GB         2 KB      0.00%
```
Then I signed the application via `west sign -t imgtool -- --key ../../../../../bootloader/mcuboot/root-rsa-2048.pem` and flash the bin/hex `west flash --bin-file ./build/zephyr/zephyr.signed.bin`,`west flash --hex-file ./build/zephyr/zephyr.signed.hex`.
```
west flash --hex-file ./build/zephyr/zephyr.signed.hex 
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner openocd
-- runners.openocd: Flashing file: build/zephyr/zephyr.signed.hex
Open On-Chip Debugger 0.11.0+dev-00726-ge2b6f5655 (2023-05-11-16:58)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
Info : DEPRECATED target event trace-config; use TPIU events {pre,post}-{enable,disable}
Info : clock speed 2000 kHz
Info : STLINK V2J40M27 (API v2) VID:PID 0483:374B
Info : Target voltage: 3.232461
Info : [stm32f7x.cpu] Cortex-M7 r0p1 processor detected
Warn : [stm32f7x.cpu] Silicon bug: single stepping may enter pending exception handler!
Info : [stm32f7x.cpu] target has 8 breakpoints, 4 watchpoints
Info : starting gdb server for stm32f7x.cpu on 3333
Info : Listening on port 3333 for gdb connections
    TargetName         Type       Endian TapName            State       
--  ------------------ ---------- ------ ------------------ ------------
 0* stm32f7x.cpu       hla_target little stm32f7x.cpu       running

Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x080022e0 msp: 0x20015ac0
Info : Unable to match requested speed 8000 kHz, using 4000 kHz
Info : Unable to match requested speed 8000 kHz, using 4000 kHz
Info : device id = 0x10016449
Info : flash size = 1024 kbytes
Warn : Adding extra erase range, 0x08040000 .. 0x0805ffff
auto erase enabled
wrote 393216 bytes from file build/zephyr/zephyr.signed.hex in 11.020806s (34.843 KiB/s)

Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
shutdown command invoke
```
I looked at the memory via `STM32CubeProgrammer`. The memory starts at `0x08000000` and the slot offset is `0x00060000` => `0x8060000`.
![memory](https://github.com/zephyrproject-rtos/zephyr/assets/20794139/b6c9c4a2-b14e-4762-b505-105ef512f32d)

@Laczen OK, but this configuration is still not working, while you have duplicated `slot1_partition`. If I removed it, then the compilation is failing due to missing `storage_partition`.
```
[142/329] Building C object zephyr/sub...ubsys__mgmt__hawkbit.dir/hawkbit.c.obj
FAILED: zephyr/subsys/mgmt/hawkbit/CMakeFiles/subsys__mgmt__hawkbit.dir/hawkbit.c.obj 
ccache /opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM7 -DHSE_VALUE=8000000 -DKERNEL -DMBEDTLS_CONFIG_FILE=\"config-tls-generic.h\" -DSTM32F756xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_ANSI_SOURCE -D_FORTIFY_SOURCE=1 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/vboxuser/zephyrproject/zephyr/include -I/home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/zephyr/include/generated -I/home/vboxuser/zephyrproject/zephyr/soc/arm/st_stm32/stm32f7 -I/home/vboxuser/zephyrproject/zephyr/lib/libc/newlib/include -I/home/vboxuser/zephyrproject/zephyr/lib/posix/getopt/. -I/home/vboxuser/zephyrproject/zephyr/drivers -I/home/vboxuser/zephyrproject/zephyr/soc/arm/st_stm32/common -I/home/vboxuser/zephyrproject/zephyr/subsys/net/l2 -I/home/vboxuser/zephyrproject/zephyr/subsys/net/lib/sockets/. -I/home/vboxuser/zephyrproject/zephyr/subsys/net/lib/dns/. -I/home/vboxuser/zephyrproject/zephyr/subsys/net/ip -I/home/vboxuser/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/vboxuser/zephyrproject/modules/hal/stm32/stm32cube/stm32f7xx/soc -I/home/vboxuser/zephyrproject/modules/hal/stm32/stm32cube/stm32f7xx/drivers/include -I/home/vboxuser/zephyrproject/modules/hal/stm32/stm32cube/stm32f7xx/drivers/include/Legacy -I/home/vboxuser/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -I/home/vboxuser/zephyrproject/modules/crypto/tinycrypt/lib/include -I/home/vboxuser/zephyrproject/modules/crypto/mbedtls/include -I/home/vboxuser/zephyrproject/zephyr/modules/mbedtls/configs -I/home/vboxuser/zephyrproject/zephyr/modules/mbedtls/include -fno-strict-aliasing -Os -imacros /home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit/build/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/vboxuser/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/vboxuser/zephyrproject/zephyr/samples/subsys/mgmt/hawkbit=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vboxuser/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vboxuser/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -D_POSIX_THREADS -std=c99 -MD -MT zephyr/subsys/mgmt/hawkbit/CMakeFiles/subsys__mgmt__hawkbit.dir/hawkbit.c.obj -MF zephyr/subsys/mgmt/hawkbit/CMakeFiles/subsys__mgmt__hawkbit.dir/hawkbit.c.obj.d -o zephyr/subsys/mgmt/hawkbit/CMakeFiles/subsys__mgmt__hawkbit.dir/hawkbit.c.obj -c /home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c
In file included from /home/vboxuser/zephyrproject/zephyr/include/zephyr/toolchain/gcc.h:92,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/toolchain.h:50,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/sys/__assert.h:11,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/sys/iterable_sections.h:10,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/logging/log_instance.h:10,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/logging/log.h:10,
                 from /home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:11:
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c: In function 'hawkbit_init':
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:85:41: error: '__device_dts_ord_DT_N_NODELABEL_storage_partition_PARENT_PARENT_ORD' undeclared (first use in this function)
   85 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:85:33: note: in expansion of macro '_CONCAT'
   85 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:211:37: note: in expansion of macro 'DEVICE_NAME_GET'
  211 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:228:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  228 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:344:9: note: in expansion of macro 'DEVICE_DT_GET'
  344 |         DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(DT_NODELABEL(label)))
      |         ^~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:59:21: note: in expansion of macro 'FIXED_PARTITION_DEVICE'
   59 | #define STORAGE_DEV FIXED_PARTITION_DEVICE(STORAGE_LABEL)
      |                     ^~~~~~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:576:27: note: in expansion of macro 'STORAGE_DEV'
  576 |         fs.flash_device = STORAGE_DEV;
      |                           ^~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:85:41: note: each undeclared identifier is reported only once for each function it appears in
   85 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:85:33: note: in expansion of macro '_CONCAT'
   85 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:211:37: note: in expansion of macro 'DEVICE_NAME_GET'
  211 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/device.h:228:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  228 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:344:9: note: in expansion of macro 'DEVICE_DT_GET'
  344 |         DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(DT_NODELABEL(label)))
      |         ^~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:59:21: note: in expansion of macro 'FIXED_PARTITION_DEVICE'
   59 | #define STORAGE_DEV FIXED_PARTITION_DEVICE(STORAGE_LABEL)
      |                     ^~~~~~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:576:27: note: in expansion of macro 'STORAGE_DEV'
  576 |         fs.flash_device = STORAGE_DEV;
      |                           ^~~~~~~~~~~
In file included from /home/vboxuser/zephyrproject/zephyr/include/zephyr/arch/arm/aarch32/arch.h:20,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/sys/cbprintf_internal.h:17,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/sys/cbprintf.h:124,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/logging/log_msg.h:11,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/vboxuser/zephyrproject/zephyr/include/zephyr/logging/log.h:11:
/home/vboxuser/zephyrproject/zephyr/include/zephyr/devicetree.h:193:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESS' undeclared (first use in this function)
  193 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/devicetree.h:4144:33: note: in definition of macro 'DT_CAT4'
 4144 | #define DT_CAT4(a1, a2, a3, a4) a1 ## a2 ## a3 ## a4
      |                                 ^~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/devicetree.h:2220:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2220 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:315:39: note: in expansion of macro 'DT_REG_ADDR'
  315 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/devicetree.h:193:29: note: in expansion of macro 'DT_CAT'
  193 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/vboxuser/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:315:51: note: in expansion of macro 'DT_NODELABEL'
  315 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:60:24: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   60 | #define STORAGE_OFFSET FIXED_PARTITION_OFFSET(STORAGE_LABEL)
      |                        ^~~~~~~~~~~~~~~~~~~~~~
/home/vboxuser/zephyrproject/zephyr/subsys/mgmt/hawkbit/hawkbit.c:582:21: note: in expansion of macro 'STORAGE_OFFSET'
  582 |         fs.offset = STORAGE_OFFSET;
      |                     ^~~~~~~~~~~~~~
[147/329] Building C object zephyr/sub...Files/subsys__net__ip.dir/dhcpv4.c.obj
ninja: build stopped: subcommand failed.
```
If I add a `storage_partition`:
```
&flash0 {
	partitions {
		compatible = "fixed-partitions";
		#address-cells = <1>;
		#size-cells = <1>;

		/* 256KB for bootloader. This is way to large but there is no way to make part of it usefull*/
		boot_partition: partition@0 {
			label = "mcuboot";
			reg = <0x00000000 DT_SIZE_K(256)>;
			read-only;
		};

		/* storage: 256KB for settings */
		storage_partition: partition@40000 {
			label = "storage";
			reg = <0x00040000 DT_SIZE_K(256)>;
		};

		/* backup slot: 256KB */
		slot0_partition: partition@80000 {
			label = "image-0";
			reg = <0x00080000 DT_SIZE_K(256)>;
		};

		/* scratch slot: 256KB */
		slot1_partition: partition@C0000 {
			label = "image-1";
			reg = <0x000C0000 DT_SIZE_K(256)>;
		};
	};
};
```
I have the same results as before
```
*** Booting Zephyr OS build v3.4.0-rc1-90-g5c7d2f1a90a6 ***
I: Starting bootloader
I: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Swap type: none
E: Image in the primary slot is not valid!
E: Unable to find bootable imag
```

--- a/boards/arm/nucleo_f756zg/nucleo_f756zg.dts
+++ b/boards/arm/nucleo_f756zg/nucleo_f756zg.dts
@@ -25,6 +25,7 @@
 		zephyr,shell-uart = &usart3;
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
 		zephyr,dtcm = &dtcm;
 	};
 
@@ -148,4 +149,41 @@ zephyr_udc0: &usbotg_fs {
 		     &eth_txd0_pg13
 		     &eth_txd1_pb13>;
 	pinctrl-names = "default";
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/*
+		 * 256KB for bootloader. This is too large but
+		 * there is no way to make the part smaller.
+		 */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(256)>;
+			read-only;
+		};
+
+		/* application image slot: 256KB */
+		slot0_partition: partition@40000 {
+			label = "image-0";
+			reg = <0x00040000 DT_SIZE_K(256)>;
+		};
+
+		/* backup slot: 256KB */
+		slot1_partition: partition@80000 {
+			label = "image-1";
+			reg = <0x00080000 DT_SIZE_K(256)>;
+		};
+
+		/* scratch slot: 256KB */
+		scratch_partition: partition@C0000 {
+			label = "image-scratch";
+			reg = <0x000C0000 DT_SIZE_K(256)>;
+		};
+
+	};
 };

